SimG4Core/Application/plugins/OscarMTProducer.cc

```diff
@@ -159,6 +159,9 @@ OscarMTProducer::OscarMTProducer(edm::ParameterSet const& p, const OscarMTMaster
   produces<edm::PCaloHitContainer>("HGCHitsEE");
   produces<edm::PCaloHitContainer>("HGCHitsHEfront");
   produces<edm::PCaloHitContainer>("HGCHitsHEback");
+  produces<edm::PCaloHitContainer>("CalibrationHGCHitsEE");
+  produces<edm::PCaloHitContainer>("CalibrationHGCHitsHEfront");
+  produces<edm::PCaloHitContainer>("CalibrationHGCHitsHEback");
```

Comment:
Just a question on this. These new calibration collections will be added to FullSim output?

Reply:
That is the idea. They will be added only when it is asked - most likely for future Phase2 versions. They will not be added when these collections are not invoked.

Reply:
Hi @bsunanda 

I am thinking on 2 things:
(1) The increased size. If I understand, it will be also in GEN-SIM, not only FEVT. Do you know the rough size of these new collections?
(2) Switching to control when they will be added. Will it introduced in later PR?

Thx.

Reply:
These collections will be needed for HGCal simulation in future. In all earlier simulations, these collections will never appear. The size will depend on type of simulation - definitely less than a few % of HGCa hit collections when they will be filled in correctly

Reply:
Currently, if I look on step1 output of Run-3 workflow in PR test, for example, [here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-a90c72/35083/runTheMatrix-results/12434.0_TTbar_14TeV+2023/), I see:

```
vector<PCaloHit>                      "g4SimHits"                 "CalibrationHGCHitsEE"   "SIM"     
vector<PCaloHit>                      "g4SimHits"                 "CalibrationHGCHitsHEback"   "SIM"     
vector<PCaloHit>                      "g4SimHits"                 "CalibrationHGCHitsHEfront"   "SIM"   
vector<PCaloHit>                      "g4SimHits"                 "HGCHitsEE"       "SIM"     
vector<PCaloHit>                      "g4SimHits"                 "HGCHitsHEback"   "SIM"     
vector<PCaloHit>                      "g4SimHits"                 "HGCHitsHEfront"   "SIM"     
```

This I expect from the way the code is, and size is tiny (I think just an empty collection). The question is that is this a good strategy to go? Or we should shut them off from before-Run-4 GEN-SIM. @civanch @cms-sw/core-l2 could you please comment? Thanks very much.